File: backend/model/src/main/java/org/example/model/entity/token/VerifyEmailToken.java
package org.example.model.entity.token;

import lombok.NoArgsConstructor;
import org.example.constant.PathnameConstants;

@NoArgsConstructor
public class VerifyEmailToken extends ConfirmationToken {
    @Override
    public void prePersist() {
        super.prePersist();
        this.pathname = PathnameConstants.VERIFY_EMAIL;
    }
}

File: laptop-store-backend/model/src/main/kotlin/org/example/model/entity/token/VerifyEmailToken.kt
package org.example.model.entity.token

import org.example.constant.PathnameConstants
import org.example.constant.TokenConstants
import javax.persistence.DiscriminatorValue
import javax.persistence.Entity
import javax.persistence.PrePersist

@Entity
@DiscriminatorValue(TokenConstants.VERIFY_EMAIL)
class VerifyEmailToken : ConfirmationToken() {
    @PrePersist
    override fun initPathnameAndTokenType() {
        this.pathname = PathnameConstants.VERIFY_EMAIL
    }
}